tests/Integration/Behaviour/Features/Context/Domain/AddressFeatureContext.php

```diff
@@ -211,14 +225,28 @@ private function updateEditCommandFields(AbstractEditAddressCommand $editAddress
         if (!empty($testCaseData['Country'])) {
             /** @var CountryByIdChoiceProvider $countryChoiceProvider */
             $countryChoiceProvider = $this->getContainer()->get('prestashop.core.form.choice_provider.country_by_id');
-            $countryId = (int) $countryChoiceProvider->getChoices()[$testCaseData['Country']];
+            $countryList = $countryChoiceProvider->getChoices();
+            if (!isset($countryList[$testCaseData['Country']])) {
+                throw new RuntimeException(sprintf(
+                    'Cannot find country %s',
+                    $testCaseData['Country']
+                ));
+            }
+            $countryId = (int) $countryList[$testCaseData['Country']];
             $editAddressCommand->setCountryId($countryId);
 
             /* @var CountryStateByIdChoiceProvider $countryStateChoiceProvider */
-            if (isset($testCaseData['State'])) {
+            if (!empty($testCaseData['State'])) {
                 $countryStateChoiceProvider = $this->getContainer()->get('prestashop.adapter.form.choice_provider.country_state_by_id');
-                $countryStateId = $countryStateChoiceProvider->getChoices(['id_country' => $countryId])[$testCaseData['State']];
-                $editAddressCommand->setStateId($countryStateId);
+                $countryStateList = $countryStateChoiceProvider->getChoices(['id_country' => $countryId]);
+                if (!isset($countryStateList[$testCaseData['State']])) {
```

Comment:
duplicate from line 356, maybe create a method to test if the State exists? :thinking:

Reply:
and line 444